src/compile/mark/text.ts

```diff
@@ -22,15 +22,33 @@ export namespace text {
     };
   }
 
-  export function properties(model: UnitModel) {
+  export function properties(model: UnitModel, referencedModel?: UnitModel) {
     // TODO Use Vega's marks properties interface
     let p: any = {};
 
     applyMarkConfig(p, model,
       ['angle', 'align', 'baseline', 'dx', 'dy', 'font', 'fontWeight',
         'fontStyle', 'radius', 'theta', 'text']);
-
+        
     const fieldDef = model.fieldDef(TEXT);
+    
+    if (referencedModel) {
+      const markType = referencedModel.mark();
```

Comment:
I modified layering to pass through all siblings just so I could get this one line (aka, what mark type is being referenced). Probably a better way?